src/hooks/useGetLogs.ts

```diff
@@ -0,0 +1,49 @@
+import { queryOptions, useQuery } from '@tanstack/react-query'
+import {
+  type BlockNumber,
+  type BlockTag,
+  type Client,
+  type GetLogsParameters,
+  stringify,
+} from 'viem'
+
+import { createQueryKey } from '~/react-query'
+
+import type { AbiEvent } from 'abitype'
+import { useClient } from './useClient'
+
+type UseLogsParameters<
+  TAbiEvent extends AbiEvent,
+  TFromBlock extends BlockNumber | BlockTag,
+  TToBlock extends BlockNumber | BlockTag,
+> = GetLogsParameters<TAbiEvent, TAbiEvent[], undefined, TFromBlock, TToBlock>
+
+export const getLogsQueryKey = createQueryKey<
+  'get-logs',
+  [key: Client['key'], args: string]
+>('get-logs')
+
+export function useGetLogsQueryOptions<
+  TAbiEvent extends AbiEvent,
+  TFromBlock extends BlockNumber | BlockTag,
+  TToBlock extends BlockNumber | BlockTag,
+>(parameters: UseLogsParameters<TAbiEvent, TFromBlock, TToBlock>) {
+  const client = useClient()
+
+  return queryOptions({
+    enabled: Boolean(parameters),
```

Comment:
We can also add `gcTime: Infinity` and `staleTime: Infinity` as we know that logs are immutable.